lib/components/Application.js

```diff
@@ -111,7 +111,7 @@ export default class Application extends Component {
         <article className="sort-filter-field">
           <h1> Shoot the Breeze </h1>
           <FilterField filterMessages={this.filterMessages.bind(this)}/>
-          <SortButtons messages={messageListItems} sortMessagesFunction={this.sortMessages.bind(this)}/>
+          <SortButtons sortMessagesFunction={this.sortMessages.bind(this)}/>
```

Comment:
Binding 'this' within render is generally frowned upon. An alternative would be to set a constructor property right after where you set state. It could look like this: **this.sortMessages = this.sortMessages.bind(this).** 

The render would look like this:  <SortButtons sortMessagesFunction={this.sortMessages}/>